app/controllers/foodcoop/ordergroups_controller.rb

```diff
@@ -1,6 +1,25 @@
 class Foodcoop::OrdergroupsController < ApplicationController
   def index
-    @ordergroups = Ordergroup.undeleted.order('name')
+    sort = if params["sort"]
+             case params["sort"]
+             when "name" then "name"
+             when "name_reverse" then "name DESC"
+             when "last_user_activity" then "max(users.last_activity)"
+             when "last_user_activity_reverse" then "max(users.last_activity) DESC"
+             when "last_order" then "max(orders.starts)"
+             when "last_order_reverse" then "max(orders.starts) DESC"
+             end
+           else
+             "name"
+           end
+
+    @ordergroups = case params["sort"]
+                   when "last_user_activity", "last_user_activity_reverse" then Ordergroup.left_joins(:users).group("groups.id")
```

Comment:
would that work with `.distinct` instead of `.group("groups.id")` too?

Reply:
No, `.distinct` doesn't seem to return the expected result.